test/trento/application/process_managers/deregistration_process_manager_test.exs

```diff
@@ -50,6 +65,49 @@ defmodule Trento.DeregistrationProcessManagerTest do
                })
     end
 
+    test "should start the process manager when DatabaseInstanceRegistered event arrives" do
+      host_id = UUID.uuid4()
+
+      assert {:start, ^host_id} =
+               DeregistrationProcessManager.interested?(%DatabaseInstanceRegistered{
+                 host_id: host_id
+               })
+    end
+
+    test "should start the process manager when ApplicationInstanceRegistered event arrives" do
+      host_id = UUID.uuid4()
+
+      assert {:start, ^host_id} =
+               DeregistrationProcessManager.interested?(%ApplicationInstanceRegistered{
+                 host_id: host_id
+               })
+    end
+
+    test "should start process managers when SapSystemRolledUp arrives" do
+      db_host_id = UUID.uuid4()
+      app_host_id = UUID.uuid4()
+
+      database_instances = [
+        build(:sap_system_instance, host_id: db_host_id)
```

Comment:
Let's be more generous here and create a list of instances. At the end, we do operations with them